#! /bin/sh -e
## 03_script.dpatch by Joerg Jaspert <joerg@debian.org>
## Original made by Eduard Bloch <blade@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Small patch to the cdda2mp3 script to read a default config.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/inet/cvs/cdrtools-2.0+a14/cdda2wav/cdda2mp3 cdrtools-2.0+a14/cdda2wav/cdda2mp3
--- /home/inet/cvs/cdrtools-2.0+a14/cdda2wav/cdda2mp3	2000-06-24 07:47:48.000000000 +0200
+++ cdrtools-2.0+a14/cdda2wav/cdda2mp3	2003-06-04 00:02:36.000000000 +0200
@@ -14,19 +14,35 @@
 
 # specify the sampling program and its options
 # do not specify the track option here!
-CDDA2WAV=cdda2wav
-CDDA2WAV_OPTS='-H -P0 -q'
+CDDA2WAV=${CDDA2WAV:-cdda2wav}
+CDDA2WAV_OPTS=${CDDA2WAV_OPTS:-'-H -P0 -q'}
 
 # for normal use, comment out the next line
 #DEBUG='-d1'
 
 # the post processor is fed through a pipe to avoid space waste
 # specify the post processing program and its options
-MP_CODER=lame
-#MP_OPTIONS=''
+MP_CODER=${MP_CODER:-lame}
+MP_OPTIONS=${MP_OPTIONS:-''}
 
+$MP_CODER -h >/dev/null 2>&1
+if [ $? != 0 ] ; then
+   echo "Encoder not found. Install one first!"
+   exit 1
+fi
+
+if [ "$CDDADEVICE" = "" ]
+then
+  CDDA_DEVICE=/dev/cdrom
+  export CDDA_DEVICE
+fi
+
 FILEPREFIX=${1:-audiotrack}
 
+if [ -e /etc/default/cdda2mp3 ]; then
+	. /etc/default/cdda2mp3
+fi
+
 TRACK=1
 while :
 do
diff -urNad /home/inet/cvs/cdrtools-2.0+a14/cdda2wav/cdda2ogg cdrtools-2.0+a14/cdda2wav/cdda2ogg
--- /home/inet/cvs/cdrtools-2.0+a14/cdda2wav/cdda2ogg	2002-04-09 13:18:15.000000000 +0200
+++ cdrtools-2.0+a14/cdda2wav/cdda2ogg	2003-06-04 00:02:33.000000000 +0200
@@ -14,26 +14,34 @@
 
 # specify the sampling program and its options
 # do not specify the track option here!
-CDDA2WAV=cdda2wav
-CDDA2WAV_OPTS='-H -P0 -q'
+CDDA2WAV=${CDDA2WAV:-cdda2wav}
+CDDA2WAV_OPTS=${CDDA2WAV_OPTS:-'-H -P0 -q'}
 
 # for normal use, comment out the next line
 #DEBUG='-d1'
 
 # the post processor is fed through a pipe to avoid space waste
 # specify the post processing program and its options
-MP_CODER=oggenc
-MP_OPTIONS=''
+MP_CODER=${MP_CODER:-oggenc}
+MP_OPTIONS=${MP_OPTIONS:-''}
 
-$MP_CODER -h > /dev/null 2> /dev/null
+$MP_CODER -h >/dev/null 2>&1
 if [ $? != 0 ] ; then
    echo "Oggenc not found. Install vorbis-tools first!"
    exit 1
 fi
 
+if [ "$CDDADEVICE" = "" ]
+then
+  CDDA_DEVICE=/dev/cdrom
+  export CDDA_DEVICE
+fi
+
 FILEPREFIX=${1:-audiotrack}
 
-. /etc/default/cdda2ogg 2>/dev/null || true
+if [ -e /etc/default/cdda2ogg ]; then
+	. /etc/default/cdda2ogg
+fi
 
 TRACK=1
 while :
